airbyte-integrations/bases/base-normalization/normalization/transform_catalog/catalog_processor.py

```diff
@@ -135,7 +135,15 @@ def build_stream_processor(
                 primary_key = get_field(configured_stream, "primary_key", f"Undefined primary key for stream {stream_name}")
 
             message = f"'json_schema'.'properties' are not defined for stream {stream_name}"
-            properties = get_field(get_field(stream_config, "json_schema", message), "properties", message)
+            schema = get_field(stream_config, "json_schema", f"'json_schema' is not defined for stream {stream_name}")
+            if "properties" in schema:
+                properties = get_field(schema, "properties", message)
+            elif "oneOf" in schema:
+                options = filter(lambda option: "properties" in option, schema["oneOf"])
+                if len(options) == 0:
+                    raise KeyError(f"Stream {stream_name} does not have any properties")
```

Comment:
Wouldn't this break for 
`{oneOf: [{type: object, properties: {...}}, {ref: string}]}` because `{ref: string}` does not have `properties` ?

Reply:
in this case, `schema["oneOf"]` would be `[{type: object, properties: {...}}, {ref: string}]`

so the `filter` call would return `[{type: object, properties: {...}}]`

and then `len(options)` would be 1

I think?